sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/DefaultVolumeSpec.java

```diff
@@ -60,6 +72,7 @@ private DefaultVolumeSpec(
         super(name, value, role, preReservedRole, principal);
         this.type = type;
         this.containerPath = containerPath;
+        this.profiles = profiles;
```

Comment:
Even after the Java code isn't passing in `null` anymore, this code should still check for `null` because it's being invoked to read preexisting configs in ZK. It could be written as e.g. `this.profiles = profiles == null ? Collections.emptyList() : profiles;`. This pattern can be seen in other `Default*Spec` types where values may be unset in the serialized rendering

Reply:
This is actually why I went with `null` instead of an empty list. But since it's more consistent to go with an empty list, I'll do what you suggested.